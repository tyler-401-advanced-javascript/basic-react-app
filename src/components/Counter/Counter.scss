.Counter {
  height: 500px;
  width: 50%;
  background-color: blanchedalmond;
  display: flex;
  flex-direction: column;
  align-items: center;
  font: {
    family: Arial;
    size: 2em;

  }
  button {
    display: block;
    margin-top: 4em;
  }
}
