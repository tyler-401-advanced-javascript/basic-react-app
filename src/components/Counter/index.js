import React from 'react'
import './Counter.scss'

class Thing extends React.Component {
  constructor () {
    super()
    this.windows = ['Green', 'Blue', 'Worst', 'Angry', 'Unhelpful', 'Rude'];
    this.styles = {
      dark: {
        backgroundColor: 'rgb(17, 20, 64)',
        color: 'rgb(227, 226, 220)'
      },
      light: {
        backgroundColor: 'rgb(245, 245, 245)',
        color: 'rgb(39, 41, 35)'
      }
    }

    this.state = {
      counter: 0,
      window: 0,
      style: true
    }
  }

  handleChange = e => {
    const changeVal = e.target.id === 'increase' ? 1 : -1;
    this.setState({
      counter: this.state.counter + changeVal,
      window: this.state.window + 1
    })
  }

  handleChangeStyle = () => {
    this.setState({
      style: !this.state.style
    })
  }

  render () {
    return (
      <section className="Counter" style={this.state.style ? this.styles.light : this.styles.dark}>
        <p>My wonderful component</p>
        <span>Now serving customer number: {this.state.counter}</span>
        <span>At window number: {this.windows[this.state.window % this.windows.length]}</span>
        <button id="increase" onClick={this.handleChange}>Take Ticket</button>
        <button id="decrease" onClick={this.handleChange}>Take Somebody Else's Ticket</button>
        <button id="style" onClick={this.handleChangeStyle}>Dark / Light</button>
      </section>
    )
  }
}

export default Thing
