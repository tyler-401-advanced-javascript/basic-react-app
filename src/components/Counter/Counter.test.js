import React from 'react'
import Counter from '.'
import renderer from 'react-test-renderer'
import { mount } from 'enzyme'

describe('<Counter />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Counter />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('changes the state on a click', () => {
    //mount  the counter.
    const component = mount(<Counter />)

    //find a #increase
    const button = component.find('#increase')
    button.simulate('click')
    
    expect(component).toHaveState({ counter: 1 })
    expect(component.find('span')).toIncludeText('1')
  })
})
