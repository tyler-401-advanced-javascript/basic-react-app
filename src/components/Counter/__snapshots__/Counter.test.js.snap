// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Counter /> renders correctly 1`] = `
<section
  className="Counter"
>
  <p>
    My wonderful component
  </p>
  <span>
    Stuff: 
    0
  </span>
  <button
    id="increase"
    onClick={[Function]}
  >
    Up
  </button>
  <button
    id="decrease"
    onClick={[Function]}
  >
    Down
  </button>
</section>
`;
